### ISSUE TITLE
[BUG] BIGTREETECH SKR PRO V1.1 - only boots when serial host connected

### ISSUE THREAD

**Issue by NickMilton** [MarlinFirmware/Marlin#14767]
### Description
BIGTREETECH SKR PRO V1.1 32 Bit Control Board running with BIGTREETECH 'RepRapDiscount Full Graphic Smart Controller" 12864 LCD. Marlin 2.0 (27 July 2019) complies and loads to the board normally. However on power up...

(1) The LCD lights up but does not display the boot screen until I plug the control board USB cable into a computer

(2) The LCD remains stuck on the Marlin boot screen until I connect a host (In this case Repetier Host) running on the computer.

Otherwise the printer operates normally and prints well. The issue exhibits itself when the board is fully wired into the printer but also when the board is connected only to the 12864 & power - so it does not seem to be a printer wiring issue. The issue is 100% reproduceable.

### Steps to Reproduce

1. Power up the BIGTREETECH SKR PRO V1.1 and BIGTREETECH 12864 with no USB cable connected  
2. Apply power and observe LCD light up but display remain blank
3. Connect SKR PRO V1.1 board USB port to a running computer and observe Marlin boot screen appear and remain on-screen indefinately
4. Launch Repetier Host on the PC and press the <connect printer> button and observe Marlin continue to boot normally as soon as the host connection is made

**Expected behavior:** The board should boot up without having to connect USB or host.

**Actual behavior:** The board does not successfully boot standalone

#### Additional Information

* Include a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files.
* Provide pictures or links to videos that clearly demonstrate the issue.
* See [How Can I Contribute](#how-can-i-contribute) for additional guidelines.

[#14767 Configs.zip](https://github.com/MarlinFirmware/Marlin/files/3441277/14767.Configs.zip)

Kind regards

**Comment by Ludy87** [comment]
ok

**Comment by golfromeo-fr** [comment]
can it be related to USB configuration of SERIAL_PORT 1/2 ? or the pin layout is not exactly the same in the 2 repositories?
https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/issues/18

~I will test my MKS LCD 12864 which doesn't light up but I haven't checked if it is better when I open pronterface.~ Until I receive my BIGTREETECH LCD 12864

edit: according to files for MKS SBase board here, MKS LCD 12864 pin mapping is not compatible with BIGTREETECH LCD 12864

**Comment by NickMilton** [comment]
Ahh

Good pick up. I set up the port configs to match the Bigtreetech provided
build - assuming they were correct. I'll try the other way round. Will not
be till weekend  - travelling.

**Comment by dch1921** [comment]
I believe the first serial port needs to be -1. The second can be 1 or 6 if you are using an esp01 wifi

**Comment by golfromeo-fr** [comment]
> I believe the first serial port needs to be -1. The second can be 1 or 6 if you are using an esp01 wifi

is it related to
https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/issues/9
?

**Comment by dch1921** [comment]
> > I believe the first serial port needs to be -1. The second can be 1 or 6 if you are using an esp01 wifi
> 
> is it related to
> [bigtreetech/BIGTREETECH-SKR-PRO-V1.1#9](https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/issues/9)
> ?

Not directly related but from what I understand Marlin uses serial port 0 or -1. When this is defined as the second serial port, Marlin doesn't fully load until a USB connection is established.

**Comment by NickMilton** [comment]
Problem solved - thanks for your suggestions golfromeo-fr and dch1921.

All I did was set....
#define SERIAL_PORT -1
#define SERIAL_PORT_2 1

and it all came good. Not a bug in Marlin - was a limitation in user knowledge. The Bigtreetech board is a good one too. No complaints there at all.

Cheers

**Comment by farmstar72** [comment]
I'm having the same problem but this fix didn't work for me - anything else I should look at?

Thanks,
Scott

**Comment by NickMilton** [comment]
Hi farmstar72

1st thing I would do is compare your config files to mine.....and see what the differences are. Could be a hardware/wiring issue?

**Comment by farmstar72** [comment]
Hi Nick, I used your config file and still have this issue. My board is exactly as shipped with no drivers installed. I did try removing the driver jumpers to no effect.

Thanks,
Scott

**Comment by NickMilton** [comment]
Scott - I 'grew up' on Repetier f/w so still llearning how Marlin behaves (so far I like it very much). I am wondering if you need to connect up thermistors, LCD etc in order for Marlin to fully boot. Just a thought. Also I have not installed the WiFi module yet. Other than that mine is stock too..

**Comment by farmstar72** [comment]
Nick, I've only been at this a few months. I started with Arduino IDE and an MKS Gen L board. Then I moved to VsCode to use Marlin 2.0 with an SKR V1.3 board. I've compiled Marlin 2.0 with VsCode/Platfomio for MKS Gen L, Ramps 1.4, Re-Arm and SKR V1.3. Each have had their challenges - ie; too long of file name, anti-virus issues, firewall, etc. - but I've made it through them all (using the same RepRapDiscount Full Graphic Smart Controller). This one has me kinda stumped though.

Thanks for your input,
Scott

**Comment by farmstar72** [comment]
WooHoo! - I got it to work!
Here's what I learned 
You have to copy the firmware to the SD card and not upload using the USB. The USB lets you upload using a Com port but with inconsistent results. I finally gave up and started my trial and error but using the SD card plus your configuration and it works now.

Thanks,
Scott

**Comment by NickMilton** [comment]
Good one. I have only ever been uploading using the microSD card - which explains our different experiences.

**Comment by farmstar72** [comment]
I've uploaded with USB to all my other boards, including 32 bit ARM's. The 32 bit ARM boards have SD cards (Re-Arm/Ramps and SKR V1.3) that show up as drive D: when I connect my USB cable and the card is inserted in the board - With those boards, I can either paste firmware.bin onto the SD or flash the board over USB. When I first got the SKR Pro, I thought it was defective because I couldn't see the SD card like that. The seller set me straight there but I was still following the habit of flashing over USB since I could. Glad I got that sorted. 
To summarize to anyone that has this same issue; (Display freezing on Marlin screen)
1. Modify config.h as above with:
#define SERIAL_PORT -1
#define SERIAL_PORT_2 1
2. You must copy and paste the firmware.bin file directly to the SD card then insert it into the SKR Pro and power up.
- Also, something interesting I noticed; Once the firmware is flashed to the board from the SD card, the firmware.bin file is deleted from the SD cardand only firmware.cur remains.
